### PR TITLE
Get back `Tabs from other devices` history section;

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
This reverts some changes from https://github.com/brave/brave-core/pull/212 and adds `More...` item into History menu for devices which have tabs number in session more than 4.

Resolves https://github.com/brave/brave-browser/issues/10473

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [x] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [x] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
1. Create sync chain with 2 devices (A and B)
2. Enable on devices `History` and `Open Tabs` sync types
3. On device A open 3 tabs, excluding internal pages (brave://****), like: google.com, apple.com, microsoft.com
4. On device B open 5 tabs, excluding internal pages (brave://****), like: cnn.com, espn.com, nytimes.com, reddit.com, ebay.com
5. On devices A and B open History page (Ctrl+H or `brave://history`) ensure there is a section `Tabs from other devices` (you may need to extend width if the width of a window is small)
6. On device A open app menu, then History sub menu, ensure you can see tabs from device B and `More...` item which leads to `brave://history/syncedTabs`
7. On device B open app menu, then History sub menu, ensure you can see 3 tabs from device A and there is no `More...` item 

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by AlexeyBarabash** [comment]
For now it looks 
Screenshot from 2020-11-11 19-02-21

Need to make an additional menu item in main menu => History => Device

**Comment by AlexeyBarabash** [comment]
The screenshot of the extended menu, note that session from phone mi4c has only 2 entries, so it doesn't have `More...` item:
Screenshot from 2020-11-17 20-22-28_2

**Comment by AlexeyBarabash** [comment]
Faced with DCHECK and CHECK:

https://source.chromium.org/chromium/chromium/src/+/master:components/sync/model_impl/processor_entity.cc;l=181;drc=f50df03b5c021e63d80ecf09ea452fa9976071ea;bpv=1;bpt=1?q=ProcessorEntity::RecordIgnoredUpdate&ss=chromium%2Fchromium%2Fsrc
```
void ProcessorEntity::RecordIgnoredUpdate(const UpdateResponseData& update) {
  DCHECK(metadata_.server_id().empty() ||
         metadata_.server_id() == update.entity.id);
```
https://source.chromium.org/chromium/chromium/src/+/master:content/public/browser/web_ui_message_handler.h;l=125;drc=f50df03b5c021e63d80ecf09ea452fa9976071ea;bpv=1;bpt=1?q=content%2Fpublic%2Fbrowser%2Fweb_ui_message_handler.h&ss=chromium%2Fchromium%2Fsrc
```
  template <typename... Values>
  void CallJavascriptFunction(const std::string& function_name,
                              const Values&... values) {
    CHECK(IsJavascriptAllowed()) << "Cannot CallJavascriptFunction before "
                                    "explicitly allowing JavaScript.";
```

Looking into this before publish the PR.

**Comment by AlexeyBarabash** [comment]
I wasn't able to see the DCHECK and CHECK mentioned in https://github.com/brave/brave-core/pull/7121#issuecomment-729174574, so opened the PR

File: app/brave_generated_resources.grd [hunk 1]
```diff
@@ -172,6 +172,9 @@
       <message name="IDS_OPEN_GUEST_PROFILE" desc="The app menu item to create a new User Profile">
         Open Guest Window
       </message>
+      <message name="IDS_OPEN_MORE_OTHER_DEVICES_SESSIONS" desc="The history sub-menu item to show sessions from other devices">
+        More...
```
Comment (petemill) [inline app/brave_generated_resources.grd:176]: Please use the ellipsis character `…`

File: chromium_src/chrome/browser/ui/toolbar/recent_tabs_sub_menu_model.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_TOOLBAR_RECENT_TABS_SUB_MENU_MODEL_H_
+#define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_TOOLBAR_RECENT_TABS_SUB_MENU_MODEL_H_
+
+#define BRAVE_RECENT_TABS_SUB_MENU_MODEL_H_ \
+  private: \
+    std::auto_ptr<sessions::SessionTab> stub_tab_; \
+  public:
+// define BRAVE_RECENT_TABS_SUB_MENU_MODEL_H_
```
Comment (petemill) [inline chromium_src/chrome/browser/ui/toolbar/recent_tabs_sub_menu_model.h:13]: remove comment?

**Review by petemill — CHANGES_REQUESTED** [review]
Minor string change

File: app/brave_generated_resources.grd (hunk 1 above)
Comment (AlexeyBarabash) [reply app/brave_generated_resources.grd]: @petemill  this is fixed at https://github.com/brave/brave-core/pull/7121/files#diff-6314ab9f5910c0e3dd53f8ac6bf09b584020abe6790abe7ebcdacc5756c5d5f9R176

File: chromium_src/chrome/browser/ui/toolbar/recent_tabs_sub_menu_model.h (hunk 2 above)
Comment (AlexeyBarabash) [reply chromium_src/chrome/browser/ui/toolbar/recent_tabs_sub_menu_model.h]: @petemill  fixed at https://github.com/brave/brave-core/pull/7121/files#diff-a996d007564a56862f193109f092f542f325c0356535e1a4b42f11e00d7143f6R13

**Comment by AlexeyBarabash** [comment]
macOS, iOS, Windows CI were aborted without any PR-specific error - restarted them all

**Comment by AlexeyBarabash** [comment]
iOS node is disabled, so can't restart.

**Review by petemill — APPROVED** [review]
Tested and works well 👍  Great to see this working 🥳

File: patches/chrome-VERSION.patch [hunk 3]
```diff
@@ -1,5 +1,5 @@
 diff --git a/chrome/VERSION b/chrome/VERSION
-index 2c4514a701725958eed9feacc7a43c46ef1f2fba..205283ea59b247e3aa5ac69d0d81f2c05133c5b8 100644
+index 2c4514a701725958eed9feacc7a43c46ef1f2fba..6d423561e1ebcc4ca78cb414346bbb09695b328f 100644
```
Comment (mkarolin) [inline patches/chrome-VERSION.patch:2]: This patch shouldn't be in the PR. No need to rerun CI if you remove it.

**Review by mkarolin — APPROVED** [review]
`chromium_src`/`patches` changes LGTM

**Comment by AlexeyBarabash** [comment]
Verified it works on Nightly.

Brave | 1.19.43 Chromium: 87.0.4280.67 (Official Build) nightly (64-bit)
-- | --
Revision | 0e5d92df40086cf0050c00f87b11da1b14580930-refs/branch-heads/4280@{#1441}

**Comment by bsclifton** [comment]
Set missing milestone

**Pull request merged**